src/hepattn/models/decoder.py

```diff
@@ -45,13 +51,15 @@ def __init__(
         decoder_layer_config["mask_attention"] = mask_attention
 
         self.decoder_layers = nn.ModuleList([MaskFormerDecoderLayer(depth=i, **decoder_layer_config) for i in range(num_decoder_layers)])
-        self.tasks = None  # Will be set by MaskFormer
+        self.tasks: list | None = None  # Will be set by MaskFormer
         self.num_queries = num_queries
         self.mask_attention = mask_attention
         self.use_query_masks = use_query_masks
         self.log_attn_mask = log_attn_mask
+        self.key_posenc = key_posenc
         self.query_posenc = query_posenc
         self.preserve_posenc = preserve_posenc
+        self.posenc_analysis = posenc_analysis
```

Comment:
What is this for? I saw `posenc_analysis_logging()` but not sure we still need it?